src/tests/network-api-test/network-api-test.c

```diff
@@ -0,0 +1,592 @@
+    UtAssert_True(actual == expected, "OS_SocketAddrGetPort() (%ld) == OS_INVALID_POINTER", (long)actual);
+
+    expected = OS_INVALID_POINTER; 
+    actual = OS_SocketAddrGetPort(&PortNum, NULL);
+    UtAssert_True(actual == expected, "OS_SocketAddrGetPort() (%ld) == OS_INVALID_POINTER", (long)actual);
+
+    /* OS_SocketGetIdByName */
+    expected = OS_INVALID_POINTER; 
+    actual = OS_SocketGetIdByName(NULL,NULL);
+    UtAssert_True(actual == expected, "OS_SocketGetIdByName() (%ld) == OS_INVALID_POINTER", (long)actual);
+
+    expected = OS_ERR_NAME_NOT_FOUND; 
+    actual = OS_SocketGetIdByName(&objid,"NF");
+    UtAssert_True(actual == expected, "OS_SocketGetIdByName() (%ld) == OS_ERR_NAME_NOT_FOUND", (long)actual);
+
+    /* OS_SocketGetInfo */
+    expected = OS_INVALID_POINTER;
+    actual = OS_SocketGetInfo(1, NULL);
+    UtAssert_True(actual == expected, "OS_SocketGetInfo() (%ld) == OS_INVALID_POINTER", (long)actual);
+
+    expected = OS_ERR_INVALID_ID;
+    actual = OS_SocketGetInfo(0, &prop);
+    UtAssert_True(actual == expected, "OS_SocketGetInfo() (%ld) == OS_ERR_INVALID_ID", (long)actual);
+
+} /* end TestDatagramNetworkApi */
+
+/*****************************************************************************
+ *
+ * Datagram Network Teardown
+ *
+ *****************************************************************************/
+void TestDatagramNetworkApi_Teardown(void)
+{
+    /* Close sockets */
+    OS_close(p1_socket_id);
+    OS_close(p2_socket_id);
+
+} /* end TestDatagramNetworkApi_Teardown */
+
+
+/*****************************************************************************
+ *
+ * Server Function for Stream Network Functional Test
+ *
+ *****************************************************************************/
+void Server_Fn(void)
+{
+    OS_SockAddr_t addr;
+    char Buf_send_s[4] = {0};
+    char Buf_rcv_s[4] = {0};
+    char Buf_expected[4] ={0};
+    int32 status;
+    uint32 connsock_id = 0;
+
+    /* Accept incoming connections */
+    status = OS_SUCCESS;
+    actual = OS_SocketAccept(s_socket_id, &connsock_id, &addr, OS_PEND);
+    UtAssert_True(status == expected, "OS_SocketAccept() (%ld) == OS_SUCCESS", (long)actual);
+    UtAssert_True(connsock_id != 0, "connsock_id (%lu) != 0", (unsigned long)connsock_id);
+
+    /* Once connected, send data to client */
+    strcpy(Buf_send_s, "abc");
+    status = sizeof(Buf_send_s);
+    actual = OS_TimedWrite(connsock_id, Buf_send_s, sizeof(Buf_send_s), OS_PEND);
+    UtAssert_True(actual == status, "OS_TimedWrite() (%ld) == %ld",(long)actual, (long)status);
+
+   /* Recieve incoming data from client*/
+    status = sizeof(Buf_rcv_s);
+    strcpy(Buf_expected, "xyz");
+
+    actual = OS_TimedRead(connsock_id, Buf_rcv_s, sizeof(Buf_rcv_s), OS_PEND);
+    UtAssert_True(actual == status, "OS_TimedRead() (%ld) == %ld",(long)actual, (long)status);
+    UtAssert_True(strcmp(Buf_rcv_s,Buf_expected) == 0, "Buf (%s) == Buf_expected (%s)", Buf_rcv_s, Buf_expected);
```

Comment:
A few issues here, first note that doing any "Assert" calls here in a child thread will cause potential overlap in the console.  For instance on my machine I get an output like this (cut and paste exactly as I see it):

```
[ PASS] 02.018 network-api-test.c:496 - Buf (abc) == Buf_expected (abc)
[ PASS][ PASS]  02.019 network-api-test.c:502 - OS_TimedWrite() (4) == 402.020 network-api-test.c:389 - OS_TimedRead() (4) == 4

[ PASS][ PASS]  02.021 network-api-test.c:512 - OS_SocketConnect() (-2) == OS_INVALID_POINTER02.022 network-api-test.c:390 - Buf (xyz) == Buf_expected (xyz)

[ PASS] 02.024 network-api-test.c:398 - OS_SocketAccept() (-2) == OS_INVALID_POINTER
```

Recommendation is to NOT do any tests in this thread, just "echo" the data back to the sender, with (possibly) a translation like ROT13 which the test can confirm.  (i.e. so "abc" becomes "nop").

Second comment is that after doing this it needs to `OS_close` the socket.

Finally worth noting that in a stream context, a single write on the sender side need not be a single read on the receiver.   4 bytes written can be read as 2+2, 3+1, etc - all valid/acceptable and not an error.